src/frontend/Dockerfile

```diff
@@ -4,14 +4,14 @@ FROM node:14-alpine
 # Set the working directory in the container
 WORKDIR /app
 
-# Copy the local directory's contents into the container at /app
-COPY . .
+COPY package.json package.json
+COPY package-lock.json package-lock.json
 
 # Install dependencies using npm or yarn
 RUN npm install
 
-# Make port 3000 available to the world outside this container
-EXPOSE 3000
+# Copy the local directory's contents into the container at /app
+COPY . .
```

Comment:
Can you explain why do we have to install dependencies before copying other things over?